### ISSUE TITLE
[Bug & a11y]: Tabs [dismissible] - keyboard interaction not possible after dismissing tab

### ISSUE THREAD

**Issue by andreancardona** [carbon-design-system/carbon#14819]
### Package

@carbon/react

### Browser

FF

### Package version

@carbon/react@1.38.0

### Automated testing tool and ruleset

Manual QA

### Assistive technology

JAWS & VO

### Description
Keyboard interaction is not possible after dismissing the current tab

After dismissing the current tab, user is unable to navigate to other active tabs using arrow keys. Note: The same issue is found in "Tab dismissible with Icon" as well.

### Recommended Fix
After dismissing the current tab, user should be able to navigate through other tabs with arrow keys.

When the standard keyboard navigation is not possible, user who is completely depended may not be able to accomplish A task.

### WCAG 2.1 Violation

2.1.1 Keyboard
